use-package.el

```diff
@@ -480,6 +480,50 @@ This is in contrast to merely setting it to 0."
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;
+;;; Little bit opinionated helper function for package.el users
+;;
+
+(defmacro use-package-with-elpa ()
+  "Set up use-package to optimal usage with package.el.
+
+For full documentation on the meaning and usage fo this, please
+consult the README file that came with this file at the section
+called `Byte-compiling with Package.el'."
+  '(progn
+     ;; Disable package initialize after us.  We either initialize it
+     ;; anyway in case of interpreted .emacs, or we don't want slow
+     ;; initizlization in case of byte-compiled .emacs.elc.
+     (setq package-enable-at-startup nil)
+     ;; Ask package.el to not add (package-initialize) to .emacs.
+     (setq package--init-file-ensured t)
+     ;; set use-package-verbose to t for interpreted .emacs,
+     ;; and to nil for byte-compiled .emacs.elc
+     (eval-and-compile
+       (setq use-package-verbose (not (bound-and-true-p byte-compile-current-file))))
+     ;; Add the macro generated list of package.el loadpaths to load-path.
+     (mapc #'(lambda (add) (add-to-list 'load-path add))
```

Comment:
Thanks for the clarification, will try to remember this!  I like the `#'` quoting a lot, because it's nice for my eyes, even if it doesn't mean anything semantically.